backend/app/services/llm/providers/gai.py

```
@@ -0,0 +1,179 @@
+        if not response.text:
+            return None, "Google AI response missing text content"
+
+        # Extract usage metadata with null checks
+        if response.usage_metadata:
+            input_tokens = response.usage_metadata.prompt_token_count or 0
+            output_tokens = response.usage_metadata.candidates_token_count or 0
+            total_tokens = response.usage_metadata.total_token_count or 0
+            reasoning_tokens = response.usage_metadata.thoughts_token_count or 0
+        else:
+            logger.warning(
+                f"[GoogleAIProvider._execute_stt] Response missing usage_metadata, using zeros"
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)